include/math/vector2d.h

```diff
@@ -0,0 +1,126 @@
+#pragma once
+
+#include "math/random.h"  // for random::*
+
+namespace kss {
+namespace math {
+
+template <typename T>
+class Vector2D {
+ public:
+  T x;
+  T y;
+
+  Vector2D() : x{0}, y{0} {}
+  Vector2D(T x, T y) : x{x}, y{y} {}
+
+  // Copy constructor and assignment operator
+  Vector2D(const Vector2D& other) : x{other.x}, y{other.y} {}
+  template <typename U>
+  Vector2D(const Vector2D<U>& other) : x{other.x}, y{other.y} {}
+
+  Vector2D& operator=(const Vector2D& other) {
+    x = other.x;
+    y = other.y;
+    return *this;
+  }
+  template <typename U>
+  Vector2D& operator=(const Vector2D<U>& other) {

```

Comment:
operator?

Reply:
Yeah, c++ supports operator overloading! These methods are the behavior that is invoked when we do assignment with the `=` operator or you can overload things like `+=`, `*`, or even `[..]`